### ISSUE TITLE
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

### ISSUE THREAD

**Issue by JoezerSmaniotto** [eslint/eslint#18825]
### Environment

Node version: 18.16.0
npm version: 9.5.1
Local ESLint version:  8.57.0
Global ESLint version:  8.57.0
Operating System: 

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
<!-- Paste your configuration here -->
```
</details>

```js
module.exports = {
  extends: ['universe', 'universe/shared/typescript-analysis'],
  overrides: [
    {
      files: ['*.ts', '*.tsx', '*.d.ts'],
      parserOptions: {
        project: './tsconfig.json',
      },
    },
  ],
  plugins: ['react-hooks'],
  rules: {
    '@typescript-eslint/consistent-type-imports': 'error',
    'import/order': [
      1,
      { groups: ['external', 'builtin', 'internal', 'sibling', 'parent', 'index'] },
    ],
    'prettier/prettier': [
      'error',
      {
        endOfLine: 'lf',
      },
    ],
    'react-hooks/rules-of-hooks': 'error', // Checks rules of Hooks
    'react-hooks/exhaustive-deps': 'warn', // Checks effect dependencies
  },
};

```

### What did you expect to happen?

When updating my react-native application that uses expo from SDK version 49 to sdk 50 when running the command "npx expo install --fix" this message appeared
" 
The following packages should be updated for best compatibility with the installed expo version:
  typescript@5.1.6 - expected version: ^5.3.0
Your project may not work correctly until you install the correct versions of the packages.
› Installing 1 SDK 50.0.0 compatible native module using npm
> npm install
"
so I updated typescript to version 5.5.4 and when running lint I get this error
"
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <5.2.0

YOUR TYPESCRIPT VERSION: 5.5.4

Please only submit bug reports when using the officially supported version.

 "

### What actually happened?

When updating my react-native application that uses expo from SDK version 49 to sdk 50 when running the command "npx expo install --fix" this message appeared
" 
The following packages should be updated for best compatibility with the installed expo version:
  typescript@5.1.6 - expected version: ^5.3.0
Your project may not work correctly until you install the correct versions of the packages.
› Installing 1 SDK 50.0.0 compatible native module using npm
> npm install
"
so I updated typescript to version 5.5.4 and when running lint I get this error
"
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <5.2.0

YOUR TYPESCRIPT VERSION: 5.5.4

Please only submit bug reports when using the officially supported version.

 "

### Link to Minimal Reproducible Example

www.google.com

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

When updating my react-native application that uses expo from SDK version 49 to sdk 50 when running the command "npx expo install --fix" this message appeared

" 
The following packages should be updated for best compatibility with the installed expo version:
  typescript@5.1.6 - expected version: ^5.3.0
Your project may not work correctly until you install the correct versions of the packages.
› Installing 1 SDK 50.0.0 compatible native module using npm
> npm install
"
so I updated typescript to version 5.5.4 and when running lint I get this error
"
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <5.2.0

YOUR TYPESCRIPT VERSION: 5.5.4

Please only submit bug reports when using the officially supported version.

 "
